c10/util/intrusive_ptr.h

```diff
@@ -595,7 +595,7 @@ class weak_intrusive_ptr final {
   }
 
   intrusive_ptr<TTarget, NullType> lock() const noexcept {
-    auto refcount = target_->refcount_.load();
+    auto refcount = use_count();
```

Comment:
Let's rather add an explicit if-test for `NullType::singleton` here because this `load()` call forms a common pattern together with the `while` loop and the `compare_exchange_weak` below. I don't want to break those up.

Reply:
I was adding a if loop to check `if (expired())` then return `NullType::singleton()`, later I realized the `use_count()` returns `load()` and return 0 for `NullType::singleton()`, which is nice to combine. But yeah if the `load()` explicitly combine with `compare_exchange_weak` to form a pattern, I can revert it back